### DISCUSSION TITLE
onClick event capture on graph elements

### DISCUSSION THREAD

**Discussion by amitjoy** [eckig/graph-editor#101] [Q&A] [answered]
Another similar question: Is there any way to capture onClick events on nodes/connections/connectors?

**Comment by eckig** [comment, marked as answer]
Have you read up on how to [add custom skins](https://github.com/eckig/graph-editor/wiki/3.-Adding-Custom-Skins)?

With your own skin added to the editor you can add any JavaFX element you like to the node and modify it in any way, like adding listeners.

Another way would be, in this special case, to track the selection with the `SelectionManager`. Each node you click (at least in the default) will toggle its selected state.
